source/module_hamilt_lcao/hamilt_lcaodft/operator_lcao/td_ekinetic_lcao.cpp

```diff
@@ -368,6 +370,19 @@ void TDEkinetic<OperatorLCAO<std::complex<double>, double>>::contributeHk(int ik
     else{        
         ModuleBase::TITLE("TDEkinetic", "contributeHk");
         ModuleBase::timer::tick("TDEkinetic", "contributeHk");
+        //Save HR_data of tddft if HR output is needed
+        if(output_hR_done <GlobalV::NSPIN && hsolver::HSolverLCAO<std::complex<double>>::out_mat_hsR == 1 && GlobalV::NSPIN!=4)
+        {
+            const int spin_now = GlobalV::CURRENT_SPIN;
```

Comment:
is there a way not to use CURRENT_SPIN ? the variable will be deleted soon

Reply:
Perhaps the function "sparse_format::cal_HContainer_cd" would be Modified or abandoned first? But yes, a small loop could be replacement.